R/markdown-link.R

```diff
@@ -174,6 +185,100 @@ parse_link <- function(destination, contents, state) {
   }
 }
 
+resolve_link_package <- function(topic, me = NULL, pkgdir = NULL) {
+  me <- me %||% roxy_meta_get("current_package")
+  # this is  from the roxygen2 tests, should not happen on a real package
+  if (is.null(me) || is.na(me) || me == "") return(NA_character_)
+
+  # if it is in the current package, then no need for package name, right?
+  if (has_topic(topic, me)) return(NA_character_)
+
+  # try packages in depends, imports, suggests first, error on name clashes
+  pkgdir <- pkgdir %||% roxy_meta_get("current_package_dir")
+  deps <- mddata[["deps"]] <- mddata[["deps"]] %||% desc::desc_get_deps(pkgdir)
```

Comment:
I found `mddata[["deps"]] <- mddata[["deps"]]` a bit hard to parse mentally.

Reply:
Yeah. But it is so much simpler than an `if () { } else {}`. Anyway, I can change it to an `if`.